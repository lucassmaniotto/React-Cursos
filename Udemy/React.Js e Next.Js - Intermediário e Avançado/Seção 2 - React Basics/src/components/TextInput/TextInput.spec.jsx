import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'test'} />);

    const input = screen.getByPlaceholderText(/what are you looking for?/i);
    expect(input.value).toBe('test');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    const value = 'lorem ipsum';

    render(<TextInput handleChange={fn} searchValue={value} />);

    const input = screen.getByPlaceholderText(/what are you looking for?/i);

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} />);
    expect(container).toMatchSnapshot();
  });
});
