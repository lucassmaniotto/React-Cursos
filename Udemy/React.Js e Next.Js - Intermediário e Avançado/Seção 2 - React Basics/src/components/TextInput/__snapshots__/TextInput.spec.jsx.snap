// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="What are you looking for?"
    type="search"
    value=""
  />
</div>
`;
